test/list/ListProxyTest.js

```diff
@@ -259,15 +260,41 @@ describe("List Proxy", function () {
         })
     });
 
+    it("listens for added and removed entry", function (done) {
+        this.timeout(5000);
+        var listener = {
+            itemAdded: function (itemEvent) {
+                expect(itemEvent.name).to.be.equal('test');
+                expect(itemEvent.item).to.be.equal(2);
+                expect(itemEvent.eventType).to.be.equal(ItemEventType.ADDED);
+                expect(itemEvent.member).to.not.be.equal(null);
+            },
+            itemRemoved: function (itemEvent) {
+                expect(itemEvent.name).to.be.equal('test');
+                expect(itemEvent.item).to.be.equal(2);
+                expect(itemEvent.eventType).to.be.equal(ItemEventType.REMOVED);
+                expect(itemEvent.member).to.not.be.equal(null);
+                done();
```

Comment:
if one of the above assertions fail, the test will never reach any `done()` or `done(err)` calls, right? Then the test will fail with a time-out rather than the real reason of assertion failure. How about this:
```
         try {
                expect(itemEvent.name).to.be.equal('test');
                expect(itemEvent.item).to.be.equal(2);
                expect(itemEvent.eventType).to.be.equal(ItemEventType.REMOVED);
                expect(itemEvent.member).to.not.be.equal(null);
                done();
          } catch(e) {
                done(e);
          }
```

Reply:
Above comment applies to all of the similar tests in this pr.

Reply:
It fails with the real reason.
Let say `expect(itemEvent.item).to.be.equal(234);`. The test fails with assertion rather than time-out.